pkg/container/reader.go

```diff
@@ -101,14 +101,12 @@ func (ctn Reader) GetToken(cid cid.Cid) (token.Token, error) {
 }
 
 // GetDelegation is the same as GetToken but only return a delegation.Token, with the right type.
+// If not found, delegation.ErrDelegationNotFound is returned.
 func (ctn Reader) GetDelegation(cid cid.Cid) (*delegation.Token, error) {
 	tkn, err := ctn.GetToken(cid)
 	if errors.Is(err, ErrNotFound) {
```

Comment:
Using `errors.Is` here implies the container can return other types of errors.  If so, why was the next code block removed?  If not, why not use `err != nil` here?

Reply:
That document the semantic ("this is the error I expected") ... but that can be a comment I suppose?

Reply:
That seems a bit fragile - if `ctn.GetToken` is later changed to return another type of error, `tkn` will be `nil` and you'll happily type that to a `*delegation.Token` on L110.

Reply:
... and then you'd silently bubble up a new type of error, even though something should likely be done about it. I'm not sure there is a perfect solution here.

Reply:
I changed anyway.